source/extensions/filters/http/custom_response/config.cc

```diff
@@ -0,0 +1,65 @@
+#include "source/extensions/filters/http/custom_response/config.h"
+
+#include <memory>
+
+#include "envoy/config/core/v3/extension.pb.validate.h"
+#include "envoy/extensions/filters/http/custom_response/v3/custom_response.pb.h"
+#include "envoy/extensions/filters/http/custom_response/v3/custom_response.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "source/common/http/matching/data_impl.h"
+#include "source/common/matcher/matcher.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace CustomResponse {
+
+namespace {
+
+class CustomResponseMatchActionValidationVisitor
+    : public Matcher::MatchTreeValidationVisitor<::Envoy::Http::HttpMatchingData> {
+public:
+  absl::Status
+  performDataInputValidation(const Matcher::DataInputFactory<::Envoy::Http::HttpMatchingData>&,
+                             absl::string_view) override {
+    return absl::OkStatus();
+  }
+};
+
+} // namespace
+
+FilterConfig::FilterConfig(
+    const envoy::extensions::filters::http::custom_response::v3::CustomResponse& config,
+    Server::Configuration::ServerFactoryContext& context, Stats::StatName stats_prefix)
+    : stats_prefix_(stats_prefix) {
+  // Allow matcher to not be set, to allow for cases where we only have route or
+  // virtual host specific configurations.
+  if (config.has_custom_response_matcher()) {
```

Comment:
Suggestion: move this `if` statement to a function in anonymous namespace that returns `Matcher::MatchTreePtr<::Envoy::Http::HttpMatchingData>` and then you can make matcher_ const. Consteness makes it easier to reason about immutability of the config after it was created.

It is too bad that the `MatchTree` does not enforce const correctness, but this is a story for another day.

Reply:
done. Missed this when making the policy config classes have only const attributes.